packages/shadow-dom/src/findInsertionPoint.ts

```diff
@@ -8,29 +8,46 @@ const styleBucketOrderingMap = styleBucketOrdering.reduce((acc, cur, j) => {
   return acc;
 }, {} as Record<StyleBucketName, number>);
 
+function isSameInsertionKey(sheetA: ExtendedCSSStyleSheet, sheetB: ExtendedCSSStyleSheet): boolean {
+  const targetKey = sheetA.bucketName + ((sheetA.metadata['m'] as string | undefined) ?? '');
+  const elementKey = sheetB.bucketName + ((sheetA.metadata['m'] as string | undefined) ?? '');
+
+  return targetKey === elementKey;
+}
+
 export function findInsertionPoint(
   renderer: GriffelShadowDOMRenderer,
-  styleSheet: ExtendedCSSStyleSheet,
+  targetStyleSheet: ExtendedCSSStyleSheet,
 ): ExtendedCSSStyleSheet | null {
   let styleSheets = renderer.adoptedStyleSheets;
-  const targetOrder = styleBucketOrderingMap[styleSheet.bucketName];
+
+  const targetOrder = styleBucketOrderingMap[targetStyleSheet.bucketName];
+  const targetPriority = (targetStyleSheet.metadata['p'] as number | undefined) ?? 0;
 
   let comparer = (sheet: ExtendedCSSStyleSheet): number => targetOrder - styleBucketOrderingMap[sheet.bucketName];
+  const priorityComparer = (sheet: ExtendedCSSStyleSheet) =>
```

Comment:
similar to this https://github.com/microsoft/griffel/pull/539#discussion_r1574997161